pkg/sql/sem/builtins/builtins.go

```diff
@@ -7299,6 +7304,222 @@ expires until the statement bundle is collected`,
+							5*time.Minute, func(ctx context.Context) error {
+								rawResp, pErr = kv.SendWrappedWithAdmission(ctx,
+									evalCtx.Txn.DB().NonTransactionalSender(), header, admissionHeader, req)
+								if pErr != nil {
+									return pErr.GoError()
+								}
+								return nil
+							})
+						// TODO(adityamaru): Think about what to do in the face of
+						// WriteIntentError. SHould we retry a few times?
+						if exportRequestErr != nil {
+							return nil, exportRequestErr
+						}
+
+						resp := rawResp.(*roachpb.ExportResponse)
+						for _, file := range resp.Files {
+							fingerprint = fingerprint ^ file.Fingerprint
+
+							// Aggregate all the range keys that need fingerprinting once all
+							// ExportRequests have been completed.
+							if len(file.SST) != 0 {
+								ssts = append(ssts, file.SST)
+							}
+						}
+						if resp.ResumeSpan != nil {
+							if !resp.ResumeSpan.Valid() {
+								return nil, errors.Errorf("invalid resume span: %s", resp.ResumeSpan)
+							}
+
+							resumeReq := req
+							resumeReq.RequestHeader = roachpb.RequestHeaderFromSpan(*resp.ResumeSpan)
+							todo <- resumeReq
+						}
+					default:
+						// No ExportRequests left to send. We've aggregated range keys
+						// across all ExportRequests and can now fingerprint them.
+						//
+						// NB: We aggregate rangekeys across ExportRequests and then
+						// fingerprint them on the client, instead of fingerprinting them as
+						// part of the ExportRequest command evaluation, because range keys
+						// do not have a stable, discrete identity. Their fragmentation can
+						// be influenced by rangekeys outside the time interval that we are
+						// fingerprinting, or by range splits. So, we need to "defragment"
+						// all the rangekey stacks we observe such that the fragmentation is
+						// deterministic on only the data we want to fingerprint in our key
+						// and time interval.
+						//
+						// Egs:
+						//
+						// t2  				[-----)[----)
+						//
+						// t1 	[----)[-----)
+						//			a			b			c			d
+						//
+						// Assume we have two rangekeys [a, c)@t1 and [b, d)@t2. They will
+						// fragment as shown in the diagram above. If we wish to fingerprint
+						// key [a-d) in time interval (t1, t2] the fragmented rangekey
+						// [a, c)@t1 is outside our time interval and should not influence our
+						// fingerprint. The iterator in `fingerprintRangekeys` will
+						// "defragment" the rangekey stacks [b-c)@t2 and [c-d)@t2 and
+						// fingerprint them as a single rangekey with bounds [b-d)@t2.
+						rangekeyFingerprint, err := fingerprintRangekeys(ctx, ssts)
+						if err != nil {
+							return nil, err
+						}
+						fingerprint = fingerprint ^ rangekeyFingerprint
+						return tree.NewDInt(tree.DInt(fingerprint)), nil
+					}
+				}
+			},
+			Info:       "This function is used only by CockroachDB's developers for testing purposes.",
```

Comment:
if this is true, then the function shouldn't appear in the public docs, and there shouldn't be a release note for it

Reply:
good point, still working through whether we want to expose this or not but will make sure to tidy it up either way.